File: python/examples/algokit_self_send.py
#!/usr/bin/env python3

import os

from algokit_utils import AlgoAmount, AlgorandClient, PaymentParams
from aplane import SignerClient
from aplane.algokit import create_apsigner_account


sender = os.environ["APLANE_ADDRESS"]
algorand = AlgorandClient.testnet()

with SignerClient.from_env() as signer:
    auth = algorand.client.algod.account_information(sender).auth_addr or sender
    account = create_apsigner_account(signer, sender, auth_address=auth)
    txn = algorand.create_transaction.payment(
        PaymentParams(
            sender=sender,
            signer=account,
            receiver=sender,
            amount=AlgoAmount(micro_algo=0),
            validity_window=1000,
        )
    )
    signed = account.signer([txn], [0])
    print(algorand.client.algod.send_raw_transaction(signed).tx_id)

File: typescript/examples/algokit_self_send.ts
import { AlgorandClient, microAlgo } from "@algorandfoundation/algokit-utils";
import { SignerClient, createApsignerAccount } from "aplane";

const sender = process.env.APLANE_ADDRESS;
if (!sender) throw new Error("APLANE_ADDRESS is required");

const algorand = AlgorandClient.testNet();
const signer = await SignerClient.fromEnv();

try {
  const info = await algorand.account.getInformation(sender);
  const account = createApsignerAccount({
    client: signer,
    address: sender,
    authAddress: info.authAddr?.toString() ?? sender,
  });
  const txn = await algorand.createTransaction.payment({
    sender,
    signer: account,
    receiver: sender,
    amount: microAlgo(0),
    validityWindow: 1000,
  });
  const signed = await account.signer([txn], [0]);
  console.log((await algorand.client.algod.sendRawTransaction(signed)).txId);
} finally {
  signer.close();
}
